aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -211,12 +210,621 @@ Tensor& replication_pad1d_backward_out_cpu_template(
+  int64_t owidth;
+
+  AT_CHECK(input.numel() > 0 && (input.dim() == 4 || input.dim() == 5),
+      "non-empty 4D or 5D (batch mode) tensor expected for input, but got: ", input);
+
+  if (input.dim() == 5)
+  {
+    dimw++;
+    dimh++;
+    dimd++;
+    dimslices++;
+  }
+
+  /* sizes */
+  nslices = input.size(dimslices);
+  idepth = input.size(dimd);
+  iheight = input.size(dimh);
+  iwidth = input.size(dimw);
+  odepth = idepth + pfront + pback;
+  oheight = iheight + ptop + pbottom;
+  owidth  = iwidth + pleft + pright;
+
+  AT_CHECK(owidth >= 1 || oheight >= 1 || odepth >= 1,
+      "input (D: ", idepth, " H: ", iheight, ", W: ", iwidth,
+      ") is too small."
+      " Calculated output D: ", odepth, " H: ", oheight, " W: ", owidth);
+
+}
+
+template <typename scalar_t>
+static void replication_pad3d_out_frame(
+    scalar_t *input_p, scalar_t *output_p,
+    int64_t nslices,
+    int64_t iwidth, int64_t iheight, int64_t idepth,
+    int64_t owidth, int64_t oheight, int64_t odepth,
+    int pleft, int pright,
+    int ptop, int pbottom,
+    int pfront, int pback)
+{
+  int iStartX = std::max(0, -pleft);
+  int iStartY = std::max(0, -ptop);
+  int iStartZ = std::max(0, -pfront);
+  int oStartX = std::max(0, pleft);
+  int oStartY = std::max(0, ptop);
+  int oStartZ = std::max(0, pfront);
+
+  int64_t k, ip_x, ip_y, ip_z;
+#pragma omp parallel for private(k, ip_x, ip_y, ip_z)
+  for (k = 0; k < nslices; k++) {
+    for (int64_t z = 0; z < odepth; z++) {
+      for (int64_t i = 0; i < oheight; i++) {
+        for (int64_t j = 0; j < owidth; j++) {
+          if (j < pleft) {
+            ip_x = pleft;
+          } else if (j >= pleft && j < iwidth + pleft) {
+            ip_x = j;
+          } else {
+            ip_x = iwidth + pleft - 1;
+          }
+          ip_x = ip_x - oStartX + iStartX;
+
+          if (i < ptop) {
+            ip_y = ptop;
+          } else if (i >= ptop && i < iheight + ptop) {
+            ip_y = i;
+          } else {
+            ip_y = iheight + ptop - 1;
+          }
+          ip_y = ip_y - oStartY + iStartY;
+
+          if (z < pfront) {
+            ip_z = pfront;
+          } else if (z >= pfront && z < idepth + pfront) {
+            ip_z = z;
+          } else {
+            ip_z = idepth + pfront - 1;
+          }
+          ip_z = ip_z - oStartZ + iStartZ;
+
+          scalar_t *dest_p = output_p + k * owidth * oheight * odepth +
+            z * owidth * oheight + i * owidth + j;
+          scalar_t *src_p = input_p + k * iwidth * iheight * idepth +
+            ip_z * iwidth * iheight + ip_y * iwidth + ip_x;
+          *dest_p = *src_p;
+        }
+      }
+    }
+  }
+}
+
+void replication_pad3d_out_cpu_template(
+    Tensor& output,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  int pleft = paddingSize[0];
+  int pright = paddingSize[1];
+  int ptop = paddingSize[2];
+  int pbottom = paddingSize[3];
+  int pfront = paddingSize[4];
+  int pback = paddingSize[5];
+  int dimw = 3;
```

Comment:
nit: These are just to be more informative when indexing, so you can mark them as `constexpr`